client/src/components/SigninButton.svelte

```diff
@@ -1,19 +1,141 @@
 <script lang="ts">
   import Modal from "./Modal.svelte";
   import { authStore as user } from "$stores/auth";
+  import { loginWithGoogle, loginWithMicrosoft, loginWithEmail, onSignOut, createUser } from "$lib/firebase/auth";
 
   // check if the user is logged in with getAuth
   let openSignInModal = false;
+  let email = "";
+  let password = "";
+  let confirmPass = "";
+  let signInButton = false;
+  let createAccountButton = false;
+  let errorMessage: string = "";
+  async function createAccount() {
+    if (password != confirmPass) {
+      password = "";
+      confirmPass = "";
+      return (errorMessage = "Passwords do not match");
+    }
+    try {
+      await createUser(email, password);
+      errorMessage = "";
+    } catch (err) {
+      return (errorMessage = err instanceof Error ? err.message : String(err));
```

Comment:
You do not need to return this and you aren't using this return value anywhere. Just assign errorMessage to the message